syntax = "proto3";

package user;

/* =========================
   Service
   ========================= */
service UserService {
  rpc GetUser (UserRequest) returns (UserResponse);
  rpc GetMovie (MovieRequest) returns (Movie);
}

/* =========================
   Requests
   ========================= */
message UserRequest {
  int32 user_id = 1;
}

message MovieRequest {
  int32 movie_id = 1;
}

/* =========================
   User Response (ตัวอย่างเดิม)
   ========================= */
message UserResponse {
  string user_name = 1;
  string email = 2;
  bool is_active = 3;
}

/* =========================
   Movie Domain
   ========================= */
message Movie {
  int32 movie_id = 1;
  string title = 2;
  int32 release_year = 3;
  string universe = 4;
  string director = 5;
  int32 duration_min = 6;
  float rating = 7;
  int64 box_office = 8;
  string description = 9;

  repeated Cast cast = 10;
  repeated Villain villains = 11;
  repeated Review reviews = 12;
}

/* =========================
   Sub Messages
   ========================= */
message Cast {
  string actor_name = 1;
  string character_name = 2;
  string role_type = 3;
}

message Villain {
  string name = 1;
  string actor = 2;
}

message Review {
  string source = 1;
  float score = 2;
}
